validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -149,21 +150,83 @@ def walk(self, address):
 
             if node.children:
                 to_process.extendleft(
-                    node.children.values())
+                    node.children)
 
     def _get(self, address):
```

Comment:
This function should probably be named `_get_or_create`, since it does quite a bit more than just getting a node.